src/java/org/apache/cassandra/transport/Dispatcher.java

```diff
@@ -99,8 +100,23 @@ public Dispatcher(boolean useLegacyFlusher)
         this.useLegacyFlusher = useLegacyFlusher;
     }
 
+    @Override
     public void dispatch(Channel channel, Message.Request request, FlushItemConverter forFlusher, Overload backpressure)
     {
+        if (!request.connection().getTracker().isRunning())
+        {
+            // We can not respond with a custom, transport, or server exceptions since, given current implementation of clients,
+            // they will defunct the connection. Without a protocol version bump that introduces an "I am going away message",
+            // we have to stick to an existing error code.
+            Message.Response response = ErrorMessage.fromException(new OverloadedException("Server is shutting down"));
+            response.setStreamId(request.getStreamId());
+            response.setWarnings(ClientWarn.instance.getWarnings());
+            response.attach(request.connection);
+            FlushItem<?> toFlush = forFlusher.toFlushItem(channel, request, response);
+            flush(toFlush);
+            return;
+        }
```

Comment:
Trying to find a test that hits this block/codepath...

Reply:
Sure: this is tested in `DisableBinaryTest#testDisallowsNewRequests`. I did try to separate this into the going away commit, but have missed separating tests. Apologies!